cloudwash/providers/aws.py

```diff
@@ -5,16 +5,26 @@
 from cloudwash.utils import dry_data
 from cloudwash.utils import echo_dry
 from cloudwash.utils import total_running_time
-
+import sys
 
 def cleanup(**kwargs):
 
     is_dry_run = kwargs["dry_run"]
     data = ['VMS', 'NICS', 'DISCS', 'PIPS', 'RESOURCES', 'STACKS']
     regions = settings.aws.auth.regions
     if "all" in regions:
-        with compute_client("aws", aws_region="us-west-2") as client:
-            regions = client.list_regions()
+        default_regions = ['us-west-2', 'cn-northwest-1', 'us-gov-west-1']
+        for region in default_regions:
+            try:
+                with compute_client("aws", aws_region=region) as client:
+                    regions = client.list_regions()
+                if regions:
+                    break
+            except Exception as e:
+                pass
+        if "all" in regions:
+            logger.error(f"Unable to retrive region list using currrent token!")
+            sys.exit(1)
```

Comment:
```suggestion
        default_regions = ['us-west-2', 'cn-northwest-1', 'us-gov-west-1']
        regions = []
        for region in default_regions:
            try:
                with compute_client("aws", aws_region=region) as client:
                    _regions = client.list_regions()
                if _regions:
                    regions += _regions
                    break
            except Exception as e:
                pass
        else:
            logger.error(f"Unable to retrive region list using currrent token!")
            sys.exit(1)
```
You should append the list of regions from all default regions to iterate over.

Also, the error should only be thrown when `break` is not hit in any cycle of the loop. That is `else`.

Reply:
> You should append the list of regions from all default regions to iterate over.

That's not what the proposed patch does, afaics, because of the `break`.
Does `.list_regions()` return all the regions, or just some?

Reply:
The .list_regions() can only retrieve the regions your account belongs. That means the default region lists(['us-west-2', 'cn-northwest-1', 'us-gov-west-1']) is isolated physically and managed by standalone IAM manage systems.